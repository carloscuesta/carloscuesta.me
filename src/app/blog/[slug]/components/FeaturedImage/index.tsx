import Image from 'next/image'

type Props = {
  image: string
  lqpiImage: string
  title: string
}

const FeaturedImage = (props: Props) => (
  <header className="relative">
    <Image
      alt={props.title}
      blurDataURL={props.lqpiImage}
      className="my-8 rounded-lg"
      height={376}
      placeholder="blur"
      src={props.image}
      width={752}
    />
  </header>
)

export default FeaturedImage
