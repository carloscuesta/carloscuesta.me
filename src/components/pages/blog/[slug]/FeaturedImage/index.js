// @flow
import React from 'react'

import styles from './styles.module.css'

type Props = {
  image: string,
  title: string
}

const FeaturedImage = (props: Props) => (
  <header className={styles.header}>
    <img
      alt={props.title}
      className={`lazyload ${styles.image}`}
      data-src={props.image}
      title={props.title}
    />
  </header>
)

export default FeaturedImage
