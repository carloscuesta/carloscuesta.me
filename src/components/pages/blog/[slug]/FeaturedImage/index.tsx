type Props = {
  image: string
  lqpiImage: string
  title: string
}

const FeaturedImage = (props: Props) => (
  <header className="relative">
    <img
      alt={props.title}
      className="lazyload lazy-blur my-8 rounded-lg"
      data-src={props.image}
      title={props.title}
      src={props.lqpiImage}
      width={752}
      height={376}
    />
  </header>
)

export default FeaturedImage
