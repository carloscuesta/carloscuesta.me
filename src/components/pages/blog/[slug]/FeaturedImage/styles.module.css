.header, .image {
  height: 40vh;
}

.header {
  background-color: var(--backgroundSurface);
}

.image {
  filter: blur(5px);
  object-fit: cover;
  opacity: 0;
  transition: filter 200ms ease-in-out, opacity 200ms ease-in-out;
  width: 100%;
}

.image:global(.lazyloaded) {
  filter: blur(0);
  opacity: 1;
}

@media (max-width: 768px) {
  .header, .image {
    height: 30vh;
  }
}

@media (max-width: 568px) {
  .header, .image {
    height: 20vh;
  }
}
