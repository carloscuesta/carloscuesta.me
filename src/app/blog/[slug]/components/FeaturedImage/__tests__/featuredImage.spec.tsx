import renderer from 'react-test-renderer'

import FeaturedImage from '../index'
import * as stubs from './stubs'

describe('FeaturedImage', () => {
  it('should match FeaturedImage component', () => {
    const wrapper = renderer.create(<FeaturedImage {...stubs.props} />)

    expect(wrapper).toMatchSnapshot()
  })
})
