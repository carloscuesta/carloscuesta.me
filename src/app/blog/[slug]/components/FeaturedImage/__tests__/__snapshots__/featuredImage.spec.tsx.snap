// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeaturedImage should match FeaturedImage component 1`] = `
<header
  className="relative"
>
  <img
    alt="post title"
    className="my-8 rounded-lg"
    data-nimg="1"
    decoding="async"
    height={376}
    loading="lazy"
    onError={[Function]}
    onLoad={[Function]}
    src="/_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fcarloscuesta%2Fimage%2Fupload%2Fv1454754838%2Fkcuq34zbo1yieygmx0y6.png&w=1920&q=75"
    srcSet="/_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fcarloscuesta%2Fimage%2Fupload%2Fv1454754838%2Fkcuq34zbo1yieygmx0y6.png&w=828&q=75 1x, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fcarloscuesta%2Fimage%2Fupload%2Fv1454754838%2Fkcuq34zbo1yieygmx0y6.png&w=1920&q=75 2x"
    style={
      {
        "backgroundImage": "url("data:image/svg+xml;charset=utf-8,%3Csvg xmlns='http%3A//www.w3.org/2000/svg' viewBox='0 0 752 376'%3E%3Cfilter id='b' color-interpolation-filters='sRGB'%3E%3CfeGaussianBlur stdDeviation='20'/%3E%3C/filter%3E%3Cimage preserveAspectRatio='none' filter='url(%23b)' x='0' y='0' height='100%25' width='100%25' href='https://res.cloudinary.com/carloscuesta/image/upload/t_post-preview-lqpi/v1454754838/kcuq34zbo1yieygmx0y6.png'/%3E%3C/svg%3E")",
        "backgroundPosition": "50% 50%",
        "backgroundRepeat": "no-repeat",
        "backgroundSize": "cover",
        "color": "transparent",
      }
    }
    width={752}
  />
</header>
`;
